flips/20211208-cadence-remove-optional-reference.md

```diff
@@ -0,0 +1,116 @@
+# Optional References to Indexed Accesses
+
+| Status        | (Approved)       |
+:-------------- |:---------------------------------------------------- |
+| **FLIP #**    | [722](https://github.com/onflow/flow/pull/722) |
+| **Author(s)** | Daniel Sainati (daniel.sainati@dapperlabs.com)       |
+| **Sponsor**   | Daniel Sainati (daniel.sainati@dapperlabs.com)       |
+| **Updated**   | 2021-12-09                                           |
+
+## Objective
+
+This removes the ability to create a reference to an optional value in Cadence by 
+creating a reference to the result of an indexed access. This was previously the 
+only way to obtain a reference to an optional, and as such this change effectively 
+removes these references from the language entirely. Instead, it allows users to 
+make the type on the right-hand side of an `as` expression an optional, and 
+allows users to create optionally-typed references in place of references
+to optional values. 
+
+## Motivation
+
+As detailed in https://github.com/onflow/cadence/issues/747, it is possible to 
+crash execution with a failed dereference that is not caught by the typechecker
```

Comment:
Maybe don't use the word "crash" here: Program execution aborts / stops, but the execution environment does not crash

```suggestion
abort execution with a failed dereference that is not caught by the type checker
```